cli/src/main.rs

```diff
@@ -0,0 +1,72 @@
+use anyhow::{bail, Result};
+use clap::{ArgEnum, Parser, Subcommand};
+use linkerd_failover_cli::check;
+
+#[derive(Parser)]
+#[clap(name = "linkerd-failover", author, version, about, long_about = None)]
+#[clap(propagate_version = true)]
+/// Manages the failover extension of Linkerd service mesh.
+struct Cli {
+    #[clap(subcommand)]
+    command: Commands,
+
+    #[clap(flatten)]
+    client: kubert::ClientArgs,
+}
+
+#[derive(ArgEnum, Clone)]
+enum OutputMode {
+    Table,
+    Json,
+}
+
+#[derive(Subcommand)]
+enum Commands {
+    /// Output commands to install the failover extension
+    Install,
+    /// Output commands to uninstall the failover extension
+    Uninstall,
+    /// Check the failover extension installation for potential problems
+    Check {
+        #[clap(long)]
+        pre: bool,
+        #[clap(arg_enum, short, long, default_value = "table")]
+        output: OutputMode,
+    },
+}
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let cli = Cli::parse();
+
+    // You can check for the existence of subcommands, and if found use their
+    // matches just as you would the top level cmd
```

Comment:
this looks like it was copy-pasted from a `clap` example...should we remove this?
```suggestion
```